#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"

WorldInfo {
  basicTimeStep 10
}
Viewpoint {
  orientation 0.015398507235809182 0.01573179912738621 -0.9997576688733746 1.5494431066395695
  position 0.7941438636863901 4.261555909012015 0.3955741158711065
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 300 300
}
Robot {
  translation 0 0 1.2
  rotation 0 -1 0 0
  children [
    DEF gps1 GPS {
      translation 0 0 -0.6
      children [
        DEF gps1 Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
          name "gps1"
        }
      ]
      name "gps1"
    }
    DEF gps5 GPS {
      translation 0 -0.25 -0.6
      children [
        DEF gps5 Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
          name "gps5"
        }
      ]
      name "gps5"
    }
    DEF gps6 GPS {
      translation 0 0.25 -0.6
      children [
        DEF gps6 Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
          name "gps6"
        }
      ]
      name "gps6"
    }
    DEF gps2 GPS {
      translation 0 0 0.6
      children [
        DEF gps2 Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
          name "gps1"
        }
      ]
      name "gps2"
    }
    DEF hingejoint1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.5 0 -0.7
      }
      device [
        DEF motor1 RotationalMotor {
          name "motor1"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.5 0 -0.7
        rotation 0 1 0 1.5706772783058502
        children [
          DEF wheel1 Shape {
            appearance PBRAppearance {
              baseColor 0.537255 0.713726 1
              metalness 0
            }
            geometry Cylinder {
              height 0.1
              radius 0.5
            }
          }
        ]
        boundingObject USE wheel1
        physics Physics {
          mass 2
        }
      }
    }
    DEF hingejoint2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.5 0 -0.7
      }
      device [
        DEF motor2 RotationalMotor {
          name "motor2"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.5 0 -0.7
        rotation 0 1 0 1.571141640854182
        children [
          DEF wheel2 Shape {
            appearance PBRAppearance {
              baseColor 0.537255 0.713726 1
              metalness 0
            }
            geometry Cylinder {
              height 0.1
              radius 0.5
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel2
        physics Physics {
          mass 2
        }
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 0.560784 0.545098 0.819608
        roughness 0.2
        metalness 0
      }
      geometry Box {
        size 0.9 0.5 2
      }
    }
  ]
  boundingObject USE body
  physics Physics {
    mass 10
  }
  controller "fuzzycontroller"
}
